### DISCUSSION TITLE
Oidc Claims for tags and branches

### DISCUSSION THREAD

**Discussion by L1ghtman2k** [community/community#55099] [Actions]
Is there a way to include source branch of the tag in oidc claims?

**Comment by hrishikesh-kocharekar-dev** [comment]
the source branch of a tag in a Git repository is not a standard claim in OIDC, so it would need to be added as a custom claim.

To add a custom claim to the OIDC ID token that includes the source branch of a tag in a Git repository, you would need to modify the OIDC provider or client to include this information in the claims. The specific steps to do this would depend on the OIDC provider or client that you are using and the programming language that it is implemented in.

As a general approach, you could modify the OIDC provider or client to query the Git repository to determine the source branch of a tag and include this information in the claims that are issued in the ID token. You could use a Git command such as `git tag --sort=-creatordate` to retrieve the list of tags in descending order of creation date and then use the `git show <tag-name>` command to retrieve the commit information for each tag, including the source branch.

Once you have retrieved the source branch information, you could include it in a custom claim in the OIDC ID token. The specific steps to do this would depend on the OIDC provider or client that you are using and the format of the claims that it supports.

It's worth noting that custom claims may not be supported by all OIDC clients or relying parties, so you may need to check the documentation of your particular client or RP to ensure that it supports custom claims. Additionally, you should be mindful of the security implications of including sensitive information such as source branch information in the ID token claims and ensure that appropriate access controls and security measures are in place to protect this information.

**Reply by L1ghtman2k** [reply]
I am aware about the fact that client can edit `aud` field to their liking, but for our case, we use it as a security measure in AWS policies,  so we can't rely on that info. (Our goal is to setup AWS trust policy to only allow known protected branches).

This is primarily a feature requests for a custom claim(https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#customizing-the-token-claims) that we can add on to our `sub` claim. Another clain that would he useful is weather the action is happening from the top of the branch
